src/wasm-lib/kcl/src/engine/conn.rs

```diff
@@ -242,6 +242,16 @@ impl EngineConnection {
                                 let mut sd = session_data2.lock().unwrap();
                                 sd.replace(session.clone());
                             }
+                            WebSocketResponse::Failure(FailureWebSocketResponse {
+                                success: _,
+                                request_id,
+                                errors,
+                            }) => {
```

Comment:
we need to add this to the session data, so it can actually return the error

Reply:
add it to responses_clone, since we have the request id, i am very confused because we def did this with kittycad.rs but not when we switched to modeling-cmds cc @adamchalmers we def want this to be in the responses so that we can actually return it

Reply:
like how its done above

Reply:
if you add it to the responses do this for the wasm one too and then we should be good

Reply:
yeah if we get the request_id, we can align it with the right call and return it as a KclError rather than aimlessly logging, which would be ideal

Reply:
we could also add a test where we try and execute a kcl file unauthed and that would make sure we never break it